AxonDeepSeg/apply_model.py

```diff
@@ -1,16 +1,93 @@
 from config import axon_suffix, myelin_suffix, axonmyelin_suffix
+import numpy as np
+import torch
 from loguru import logger
+from typing import List, Literal
 
-from ivadomed import inference as imed_inference
+from nnunetv2.inference.predict_from_raw_data import nnUNetPredictor
 
-default_overlap = 48
+def setup_environment_vars():
+    '''Sets up dummy env vars so that nnUNet does not complain.'''
+    os.environ['nnUNet_raw'] = 'UNDEFINED'
+    os.environ['nnUNet_results'] = 'UNDEFINED'
+    os.environ['nnUNet_preprocessed'] = 'UNDEFINED'
+
+def split_nnunet_output():
+    '''
+    nnU-Net outputs a single mask with all classes; this function reads the 
+    model config to fetch the class names and splits the mask accordingly.
+    
+    '''
+    pass
 
 def axon_segmentation(
+                    path_inputs: List[Path],
+                    path_model: Path,
+                    model_type: Literal['light', 'ensemble']='light',
+                    gpu_id: int=-1,
+                    verbosity_level: int=0,
+                    ):
+    '''
+    Segment images by applying a nnU-Net pretrained model.
+
+    Parameters
+    ----------
+    path_inputs : List[pathlib.Path]
+        List of images to segment. We assume they all exist.
+    path_model : pathlib.Path
+        Path to the folder of the nnU-Net pretrained model. We assume it exists.
+    model_type : Literal['light', 'ensemble'], optional
+        Type of model, by default 'light'.
+    gpu_id : int, optional
+        GPU ID to use for cuda acceleration. -1 to use CPU, by default -1.
+    verbosity_level : int, optional
+        Level of verbosity, by default 0.
+    '''
+
+    setup_environment_vars()
+    # find all available folds
+    if model_type == 'light':
+        folds_avail = ['all']
+    else:
+        folds_avail = [int(str(f).split('_')[-1]) for f in path_model.glob('fold_*')]
+
+    # instantiate predictor
+    predictor = nnUNetPredictor(
+        perform_everything_on_gpu=True if gpu_id < 0 else False,
+        device=torch.device('cuda', gpu_id) if gpu_id >= 0 else torch.device('cpu'),
+    )
+    logger.info('Running inference on device: {}'.format(predictor.device))
+
+    # find checkpoint name (identical for all folds)
+    chkpt_name = next((path_model / f'fold_{folds_avail[0]}').glob('*.pth')).name
+    # init network architecture and load checkpoint
+    predictor.initialize_from_trained_model_folder(
+        str(path_model),
+        use_folds=folds_avail,
+        checkpoint_name=chkpt_name,
+    )
+    logger.info('Model successfully loaded.')
+
+    # create input list
+    input_list = [ [str(p)] for p in path_inputs]
+    output_list = [ str(p.with_suffix('')) + '_seg' for p in path_inputs ]
+
+    predictor.predict_from_files(
+        list_of_lists_or_source_folder=input_list,
+        output_folder_or_list_of_truncated_output_files=output_list,
+        save_probabilities=False,
+        overwrite=False,
+    )
```

Comment:
damn this is something I didn't expect... we trained all nnunet models on grayscale images, meaning they have a single channel. While trying `axon_segmentation()`, I encountered an image that, although looking black and white, was saved in RGB format (meaning 3 channels). This makes nnunet fail directly and unfortunately, from my experience, this is a very common occurence.

We thus have 2 options:

1. before inference, read all images and if they are in RGB format, save a grayscale conversion in PNG format + update input filenames list
2. use `nnUNetPredictor.predict_from_list_of_npy_arrays` instead of `nnUNetPredictor.predict_from_list_of_files`. I don't link this option because it adds a lot of complexity to this PR: we will need to manage the RAM on our side and it will be a problem for large number of inputs. By comparison, I currently use `predict_from_list_of_files` which is memory efficient.